# Winning Jeopardy

This a Guided Project from Dataquest.io.

The goal here is to analyze the questions and answers from the american TV show "Jeopardy" to look for patterns in order to help one preparing for it.

The database consists of 20,000 questions, which are some of the questions asked from 1984 to 2012.

In [1]:
#Initializing imports
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
jeopardy = pd.read_csv("jeopardy.csv")
jeopardy.head(10)

Show Number    Air Date      Round                         Category  Value  \
0         4680  2004-12-31  Jeopardy!                          HISTORY   $200   
1         4680  2004-12-31  Jeopardy!  ESPN's TOP 10 ALL-TIME ATHLETES   $200   
2         4680  2004-12-31  Jeopardy!      EVERYBODY TALKS ABOUT IT...   $200   
3         4680  2004-12-31  Jeopardy!                 THE COMPANY LINE   $200   
4         4680  2004-12-31  Jeopardy!              EPITAPHS & TRIBUTES   $200   
5         4680  2004-12-31  Jeopardy!                   3-LETTER WORDS   $200   
6         4680  2004-12-31  Jeopardy!                          HISTORY   $400   
7         4680  2004-12-31  Jeopardy!  ESPN's TOP 10 ALL-TIME ATHLETES   $400   
8         4680  2004-12-31  Jeopardy!      EVERYBODY TALKS ABOUT IT...   $400   
9         4680  2004-12-31  Jeopardy!                 THE COMPANY LINE   $400   

                                            Question          Answer  
0  For the last 8 years of his life, Galileo was ...      Copernicus  
1  No. 2: 1912 Olympian; football star at Carlisl...      Jim Thorpe  
2  The city of Yuma in this state has a record av...         Arizona  
3  In 1963, live on "The Art Linkletter Show", th...      McDonald's  
4  Signer of the Dec. of Indep., framer of the Co...      John Adams  
5  In the title of an Aesop fable, this insect sh...         the ant  
6  Built in 312 B.C. to link Rome & the South of ...  the Appian Way  
7  No. 8: 30 steals for the Birmingham Barons; 2,...  Michael Jordan  
8  In the winter of 1971-72, a record 1,122 inche...      Washington  
9  This housewares store was named for the packag...  Crate & Barrel

Checking unique values for some columns to look for patterns and to get familiar with the Database

In [3]:
jeopardy['Show Number'].unique()

array([4680, 5957, 3751, 3673, 4931, 5690, 2825, 6037, 5243, 3036, 4107,
       2735, 5851, 4335, 4541, 3834, 3619, 3534, 4085, 1279, 1302, 4936,
       3632, 3362, 4960, 5392, 5795, 6118, 5580, 4699, 3113, 3447, 6294,
       3461, 1800, 5981, 3112, 5431, 5630, 4595, 3214, 5145, 5282, 4220,
       6125, 5979, 3815,  422, 1740, 3697, 5079, 4487, 5807, 5622, 5084,
       5002, 5334, 5347, 4751, 4576, 4582, 5352, 4398, 6144, 6239, 5796,
       5996, 2818, 5797, 4213, 4278, 3403, 3436, 1276, 4700, 6288, 5512,
       3060, 5419, 5686, 3343, 3331, 3003,  308, 3480, 4842, 1658, 3254,
       5903, 5914, 2047, 2349, 5173, 3537, 3911, 4254, 5831, 3470, 4506,
       2120, 5935, 3903, 3303, 4968, 5307, 3517,  776, 3010, 3855, 3358,
       4985, 6287, 5930, 4306, 5363,  958, 4667,  732, 5476, 5171, 3191,
       2515, 4183, 3255, 5094, 4644, 6072, 4808, 5467, 5146, 4146, 3501,
       1274, 3547, 4227, 2679, 4724, 3898, 1291, 4946, 4787, 6135, 5443,
         62, 6231, 3682, 3102, 5397, 6293, 5624, 33

In [4]:
jeopardy.iloc[:,1].unique()

array(['2004-12-31', '2010-07-06', '2000-12-18', '2000-07-19',
       '2006-02-06', '2009-05-08', '1996-12-06', '2010-12-07',
       '2007-05-30', '1997-11-10', '2002-06-11', '1996-06-21',
       '2010-02-08', '2003-06-06', '2004-05-10', '2001-04-12',
       '2000-05-04', '2000-01-06', '2002-05-10', '1990-03-08',
       '1990-04-10', '2006-02-13', '2000-05-23', '1999-03-30',
       '2006-03-17', '2008-02-05', '2009-11-20', '2011-03-30',
       '2008-12-05', '2005-01-27', '1998-02-25', '1999-09-07',
       '2012-01-19', '1999-09-27', '1992-05-29', '2010-09-20',
       '1998-02-24', '2008-03-31', '2009-02-13', '2004-07-23',
       '1998-07-16', '2007-01-12', '2007-07-24', '2002-12-27',
       '2011-04-08', '2010-09-16', '2001-03-16', '1986-04-22',
       '1992-03-06', '2000-10-03', '2006-10-12', '2004-02-24',
       '2009-12-08', '2009-02-03', '2006-10-19', '2006-05-16',
       '2007-11-15', '2007-12-04', '2005-04-11', '2004-06-28',
       '2004-07-06', '2007-12-11', '2003-10-22', '2011-

In [5]:
jeopardy.columns

Index(['Show Number', ' Air Date', ' Round', ' Category', ' Value',
       ' Question', ' Answer'],
      dtype='object')

In [6]:
#Correcting column names
jeopardy.columns = ['Show Number', 'Air Date', 'Round', 'Category', 'Value', 'Question', 'Answer']

In [7]:
air_dates = jeopardy['Air Date'].unique()
pd.Series(air_dates).sort_values()

323    1984-09-21
143    1984-12-04
83     1985-11-13
47     1986-04-22
301    1986-04-23
294    1987-11-09
117    1987-11-10
106    1988-01-11
115    1988-11-02
242    1989-05-24
230    1989-11-06
277    1989-12-01
132    1990-03-01
73     1990-03-05
19     1990-03-08
138    1990-03-26
20     1990-04-10
299    1990-05-17
239    1990-11-15
270    1991-04-01
86     1991-11-13
48     1992-03-06
34     1992-05-29
198    1992-10-30
265    1992-12-16
90     1993-06-29
99     1993-11-19
233    1993-11-25
211    1994-11-07
91     1994-11-17
          ...    
266    2010-06-21
1      2010-07-06
45     2010-09-16
35     2010-09-20
319    2010-09-27
66     2010-10-11
186    2010-11-04
333    2010-11-19
175    2010-11-25
7      2010-12-07
324    2011-01-21
126    2011-01-25
163    2011-02-16
27     2011-03-30
44     2011-04-08
141    2011-04-22
63     2011-05-05
168    2011-05-26
275    2011-05-30
289    2011-07-12
156    2011-07-13
226    2011-07-18
206    2011-10-21
144    2011-10-24
64     201

In [8]:
jeopardy['Round'].unique()

array(['Jeopardy!', 'Double Jeopardy!', 'Final Jeopardy!', 'Tiebreaker'],
      dtype=object)

In [9]:
jeopardy['Value'].unique()

array(['$200', '$400', '$600', '$800', '$2,000', '$1000', '$1200',
       '$1600', '$2000', '$3,200', 'None', '$5,000', '$100', '$300',
       '$500', '$1,000', '$1,500', '$1,200', '$4,800', '$1,800', '$1,100',
       '$2,200', '$3,400', '$3,000', '$4,000', '$1,600', '$6,800',
       '$1,900', '$3,100', '$700', '$1,400', '$2,800', '$8,000', '$6,000',
       '$2,400', '$12,000', '$3,800', '$2,500', '$6,200', '$10,000',
       '$7,000', '$1,492', '$7,400', '$1,300', '$7,200', '$2,600',
       '$3,300', '$5,400', '$4,500', '$2,100', '$900', '$3,600', '$2,127',
       '$367', '$4,400', '$3,500', '$2,900', '$3,900', '$4,100', '$4,600',
       '$10,800', '$2,300', '$5,600', '$1,111', '$8,200', '$5,800',
       '$750', '$7,500', '$1,700', '$9,000', '$6,100', '$1,020', '$4,700',
       '$2,021', '$5,200', '$3,389'], dtype=object)

In [10]:
#Converting values to numbers
jeopardy['Value'] = jeopardy['Value'].str.replace('$','')
jeopardy['Value'] = jeopardy['Value'].str.replace(',','')
jeopardy['Value'] = jeopardy['Value'].str.replace('None','0')
jeopardy['Value'] = jeopardy['Value'].apply(float)
jeopardy['Value'].unique()

array([  200.,   400.,   600.,   800.,  2000.,  1000.,  1200.,  1600.,
        3200.,     0.,  5000.,   100.,   300.,   500.,  1500.,  4800.,
        1800.,  1100.,  2200.,  3400.,  3000.,  4000.,  6800.,  1900.,
        3100.,   700.,  1400.,  2800.,  8000.,  6000.,  2400., 12000.,
        3800.,  2500.,  6200., 10000.,  7000.,  1492.,  7400.,  1300.,
        7200.,  2600.,  3300.,  5400.,  4500.,  2100.,   900.,  3600.,
        2127.,   367.,  4400.,  3500.,  2900.,  3900.,  4100.,  4600.,
       10800.,  2300.,  5600.,  1111.,  8200.,  5800.,   750.,  7500.,
        1700.,  9000.,  6100.,  1020.,  4700.,  2021.,  5200.,  3389.])

In [11]:
#Now checking some weird values for the rounds to see if they are outliers.
jeopardy['Value'].value_counts().sort_index()

0.0         336
100.0       804
200.0      2784
300.0       764
367.0         1
400.0      3892
500.0       798
600.0      1890
700.0        15
750.0         1
800.0      2980
900.0         6
1000.0     1980
1020.0        1
1100.0        6
1111.0        1
1200.0     1111
1300.0        6
1400.0       20
1492.0        1
1500.0       50
1600.0     1046
1700.0        1
1800.0       22
1900.0        5
2000.0     1223
2021.0        1
2100.0        2
2127.0        1
2200.0       11
           ... 
3500.0        6
3600.0        8
3800.0        2
3900.0        1
4000.0       32
4100.0        1
4400.0        2
4500.0        1
4600.0        2
4700.0        1
4800.0        2
5000.0       23
5200.0        1
5400.0        1
5600.0        2
5800.0        1
6000.0        7
6100.0        1
6200.0        1
6800.0        1
7000.0        7
7200.0        2
7400.0        1
7500.0        1
8000.0        3
8200.0        1
9000.0        1
10000.0       3
10800.0       1
12000.0       2
Name: Value, Length: 72,

About the values column, it seems that, even though round numbers are more frequent, there are some occurencies of non-round values, which means it may not be safe to group them as if they were typos...

Now Dataquest is recommending to create a function that normalizes words. Seems like a good idea in order to group and analyze them later.

In [12]:
import string
#Defining function
def normalize_string(s):
    s_ = s
    s_ = str.lower(s_)
    s_ = s_.translate(str.maketrans('', '', string.punctuation))
    return s_

#A little test
normalize_string("Don't you EVER do. This. Again.")

'dont you ever do this again'

In [13]:
#Creating two Series with cleaned Question and Answer columns (I may still need the original columns, with punctuation)
clean_question = jeopardy['Question'].apply(normalize_string)
clean_answer = jeopardy['Answer'].apply(normalize_string)

In [14]:
clean_question.head(10)

0    for the last 8 years of his life galileo was u...
1    no 2 1912 olympian football star at carlisle i...
2    the city of yuma in this state has a record av...
3    in 1963 live on the art linkletter show this c...
4    signer of the dec of indep framer of the const...
5    in the title of an aesop fable this insect sha...
6    built in 312 bc to link rome  the south of ita...
7    no 8 30 steals for the birmingham barons 2306 ...
8    in the winter of 197172 a record 1122 inches o...
9    this housewares store was named for the packag...
Name: Question, dtype: object

In [15]:
clean_answer.head(10)

0        copernicus
1        jim thorpe
2           arizona
3         mcdonalds
4        john adams
5           the ant
6    the appian way
7    michael jordan
8        washington
9     crate  barrel
Name: Answer, dtype: object

In [16]:
#Converting Air Date to Datetime
jeopardy['Air Date'] = pd.to_datetime(jeopardy['Air Date'])
jeopardy['Air Date'].head(10)

0   2004-12-31
1   2004-12-31
2   2004-12-31
3   2004-12-31
4   2004-12-31
5   2004-12-31
6   2004-12-31
7   2004-12-31
8   2004-12-31
9   2004-12-31
Name: Air Date, dtype: datetime64[ns]

A good hint, suggested by Dataquest, is to see if the answer can be deducted from the question itself.

In [17]:
#Creating function to check if the answer is in the question
def answer_in_question(q, a):
    return a in q

#Testing it a bit
answer_in_question("dont you ever do something right","right")

True

In [18]:
#Now creating a Dataframe only with questions, answers and checks
qa_df = pd.DataFrame({'Question':clean_question, 'Answer':clean_answer})
qa_df.head(10)

Question          Answer
0  for the last 8 years of his life galileo was u...      copernicus
1  no 2 1912 olympian football star at carlisle i...      jim thorpe
2  the city of yuma in this state has a record av...         arizona
3  in 1963 live on the art linkletter show this c...       mcdonalds
4  signer of the dec of indep framer of the const...      john adams
5  in the title of an aesop fable this insect sha...         the ant
6  built in 312 bc to link rome  the south of ita...  the appian way
7  no 8 30 steals for the birmingham barons 2306 ...  michael jordan
8  in the winter of 197172 a record 1122 inches o...      washington
9  this housewares store was named for the packag...   crate  barrel

In [19]:
#Inserting a new column with the checks
qa_df['Answer in Question'] = qa_df.apply(lambda qa_df: answer_in_question(str(qa_df['Question']),str(qa_df['Answer'])), axis=1)
qa_df.head(10)

Question          Answer  \
0  for the last 8 years of his life galileo was u...      copernicus   
1  no 2 1912 olympian football star at carlisle i...      jim thorpe   
2  the city of yuma in this state has a record av...         arizona   
3  in 1963 live on the art linkletter show this c...       mcdonalds   
4  signer of the dec of indep framer of the const...      john adams   
5  in the title of an aesop fable this insect sha...         the ant   
6  built in 312 bc to link rome  the south of ita...  the appian way   
7  no 8 30 steals for the birmingham barons 2306 ...  michael jordan   
8  in the winter of 197172 a record 1122 inches o...      washington   
9  this housewares store was named for the packag...   crate  barrel   

   Answer in Question  
0               False  
1               False  
2               False  
3               False  
4               False  
5               False  
6               False  
7               False  
8               False  
9               False

In [20]:
qa_df['Answer in Question'].unique()

array([False,  True])

In [21]:
#So some of them were true after all..
pd.set_option('display.max_colwidth', -1)
pd.set_option('display.max_rows', 500)
qa_df[qa_df['Answer in Question']]

Question  \
46     in 2004 united launched this new service that features low fares  more seats per plane                               
266    ljubljana bratislava barcelona                                                                                       
272    istanbul ottawa amman                                                                                                
278    sofia sarajevo saigon                                                                                                
284    bucharest bonn bern                                                                                                  
290    belize city guatemala city panama city                                                                               
1137   of a 25th 30th or 40th anniversary what peanuts is celebrating in 1990                                               
1263   a brave french soldier might receive the award known as the croix de guerre meaning cross of this                    
1688   selfcongratulations egotists  this word made it all the way to no 10                                                 
1694   no 2 this verb form is a homophone of a letter of the alphabet                                                       
1840   of a pogo stick injury a dense winter fog or the bite of a comic strip possum what a pogonip is                      
2568   in 1968 the mpaa rated a film adult by giving it this many xs                                                        
2572   of 84 184 or 284 the length in years of one orbit by uranus around the sun                                           
2818   in 1842 richard owen coined this word for “dinosaur”                                                                 
3061   in physics notation the speed of light is symbolized by this letter in lower case                                    
3225   hell heaven or limbo                                                                                                 
3231   vow knight or grail                                                                                                  
3237   happy snappy or pappy                                                                                                
3243   cozy wax or quilt                                                                                                    
3249   fad dad or glad                                                                                                      
4163   of an artist a fish or a wasp its what a mud dauber is                                                               
4205   of inka dinka or doo an actual language spoken in southern sudan                                                     
4797   you do the math 40 degrees on the fahrenheit temperature scale equals this on the celsius scale                      
4872   of fireeating live pigeonshooting or water buffalo polo the one that was a 1900 olympic event                        
4984   singleletter chemical designation of a diamond                                                                       
5010   of the tree of life tree of knowledge or the tree of righteousness the 1 forbidden to adam                           
5051   of 6 10 or 14 the age of michael kearney in 1994 when he became the usas youngest college graduate                   
5162   of 1769 1869 or 1969 the year princeton began to admit women as undergraduates                                       
5596   this 10th anniversary present is present in the name of a 20th anniversary gift  platinum                            
5998   of chairs chicken or chinos what youre most likely to haul if you drive a reefer truck                               
6236   capable succeed accomplish                                                                                           
6242   paint brush easel                                                                                         

In the vast majority of positives, either the answers were too short, meaning a false positive for the string checking, or the questions were clearly of multiple choices, which obviously results in a positive.
Doing a manual count (144 rows allows me to do that), seems like only 6 go into this rule (lines 10957, 10963, 11428, 12199, 15396 and 15943), which accounts for only 0,03% of al questions.
Conclusion: there's no statistically significant relation at all.

Now checking if topics repeat themselves by checking how many times words with more than 6 characters repeat in different questions.

In [32]:
word_frequency = {}
for r in clean_question:
    r_words = list(set(r.split(' ')))
    for w in r_words:
        if len(w) > 6:
            if w not in word_frequency:
                word_frequency[w] = 1
            else:
                word_frequency[w] += 1

In [40]:
import operator
sorted_words = sorted(word_frequency.items(), key=operator.itemgetter(1), reverse=True)
sorted_words

[('country', 473),
 ('american', 250),
 ('president', 249),
 ('capital', 247),
 ('targetblankherea', 243),
 ('national', 179),
 ('largest', 176),
 ('british', 164),
 ('meaning', 160),
 ('century', 156),
 ('musical', 152),
 ('company', 151),
 ('between', 143),
 ('founded', 141),
 ('reports', 141),
 ('character', 140),
 ('include', 138),
 ('targetblankthisa', 138),
 ('million', 120),
 ('popular', 119),
 ('because', 110),
 ('through', 104),
 ('classic', 103),
 ('italian', 99),
 ('another', 97),
 ('william', 94),
 ('english', 94),
 ('countrys', 93),
 ('america', 93),
 ('published', 92),
 ('created', 90),
 ('version', 90),
 ('university', 87),
 ('england', 85),
 ('general', 85),
 ('nickname', 83),
 ('married', 80),
 ('against', 80),
 ('species', 78),
 ('college', 78),
 ('introduced', 77),
 ('original', 75),
 ('including', 75),
 ('language', 75),
 ('mountain', 74),
 ('playing', 74),
 ('magazine', 74),
 ('targetblanksarah', 73),
 ('something', 73),
 ('inspired', 72),
 ('spanish', 72),
 ('anci

From this list, we can get some topics that repeat themselves more frequently in questions. Country is by far the most repeated (2.4% of all our dictionary questions has it), with american, president and capital being very repeated too, which are mostly related to geopolitics questions.

In [43]:
#Now checking how many questions doesn't have any words that repeat 20, 50 and 100 times or more (0.1% of all questions)
low_frequency_questions_20 = 0
low_frequency_questions_50 = 0
low_frequency_questions_100 = 0
for r in clean_question:
    r_words = list(set(r.split(' ')))
    max_w_freq = 0
    for w in r_words:
        if len(w) > 6:
            if word_frequency[w] > max_w_freq:
                max_w_freq = word_frequency[w]
    if max_w_freq < 20:
        low_frequency_questions_20 += 1
        low_frequency_questions_50 += 1
        low_frequency_questions_100 += 1
    elif max_w_freq < 50:
        low_frequency_questions_50 += 1
        low_frequency_questions_100 += 1
    elif max_w_freq < 100:
        low_frequency_questions_100 += 1
print("Lower than 20 words: " + str(low_frequency_questions_20/len(clean_question)))
print("Lower than 50 words: " + str(low_frequency_questions_50/len(clean_question)))
print("Lower than 100 words: " + str(low_frequency_questions_100/len(clean_question)))

Lower than 20 words: 0.41007050352517627
Lower than 50 words: 0.6309815490774539
Lower than 100 words: 0.8204910245512276


So, there's 18% chance that a question will involve one of the terms of our dictionary with 100 or more repetitions. This chance grows to 37% when it comes to words appearing 50 or more times and to 59% when with words appearing 20 or more times, meaning that only 41% of all questions can be considered truly original (only with words that appear in less than 0.1% of all questions).

Now evaluating whether some of these words happen more in high-valued questions than lower ones.

In [44]:
jeopardy_low = len(jeopardy[jeopardy['Value']<800])
jeopardy_low

11285

In [45]:
jeopardy_high = len(jeopardy[jeopardy['Value']>=800])
jeopardy_high

8714

In [46]:
jeopardy_high_expected = jeopardy_high/len(jeopardy)
jeopardy_high_expected

0.43572178608930445

In [47]:
#Selecting a list with only the most frequent words (>=100)
highest_words = sorted_words[:23]
highest_words

[('country', 473),
 ('american', 250),
 ('president', 249),
 ('capital', 247),
 ('targetblankherea', 243),
 ('national', 179),
 ('largest', 176),
 ('british', 164),
 ('meaning', 160),
 ('century', 156),
 ('musical', 152),
 ('company', 151),
 ('between', 143),
 ('founded', 141),
 ('reports', 141),
 ('character', 140),
 ('include', 138),
 ('targetblankthisa', 138),
 ('million', 120),
 ('popular', 119),
 ('because', 110),
 ('through', 104),
 ('classic', 103)]

In [48]:
#Seems like I'll have to work with qa_df first to check if a question is low or high valued
qa_df["High"] = (jeopardy["Value"] >= 800)
qa_df

Question  \
0      for the last 8 years of his life galileo was under house arrest for espousing this mans theory                                                                                                                                                                                                                                                
1      no 2 1912 olympian football star at carlisle indian school 6 mlb seasons with the reds giants  braves                                                                                                                                                                                                                                         
2      the city of yuma in this state has a record average of 4055 hours of sunshine each year                                                                                                                                                                                                                                                       
3      in 1963 live on the art linkletter show this company served its billionth burger                                                                                                                                                                                                                                                              
4      signer of the dec of indep framer of the constitution of mass second president of the united states                                                                                                                                                                                                                                           
5      in the title of an aesop fable this insect shared billing with a grasshopper                                                                                                                                                                                                                                                                  
6      built in 312 bc to link rome  the south of italy its still in use today                                                                                                                                                                                                                                                                       
7      no 8 30 steals for the birmingham barons 2306 steals for the bulls                                                                                                                                                                                                                                                                            
8      in the winter of 197172 a record 1122 inches of snow fell at rainier paradise ranger station in this state                                                                                                                                                                                                                                    
9      this housewares store was named for the packaging its merchandise came in  was first displayed on                                                                                                                                                                                                                                             
10     and away we go                                                                                                                                                                                                                                                                                                                                
11     cows regurgitate this from the first stomach to the mouth  chew it again                                                                                                                                                   

In [79]:
highest_words_observed = {}
for r in highest_words:
    w = r[0]
    count_low = 0
    count_high = 0
    for r2 in qa_df.iterrows():
        r2_columns = str(r2[1]).split(chr(10))
        r2_words = str(r2_columns[0]).split(' ')
        while '' in r2_words:
            r2_words.remove('')
        r2_value = str(r2_columns[3]).split(' ')
        while '' in r2_value:
            r2_value.remove('')
        #print(r2_value)
        if w in r2_words:
            if r2_value[1] == "False":
                count_low += 1
            else:
                count_high += 1
    highest_words_observed[w] = count_high/(count_high+count_low)
    print(highest_words_observed[w])
highest_words_observed

0.4418604651162791
0.44
0.40562248995983935
0.38866396761133604
0.6213991769547325
0.45251396648044695
0.3977272727272727
0.4634146341463415
0.50625
0.5
0.375
0.47019867549668876
0.46153846153846156
0.475177304964539
0.6170212765957447
0.4
0.4855072463768116
0.6956521739130435
0.44166666666666665
0.3277310924369748
0.38181818181818183
0.4519230769230769
0.36893203883495146


{'country': 0.4418604651162791,
 'american': 0.44,
 'president': 0.40562248995983935,
 'capital': 0.38866396761133604,
 'targetblankherea': 0.6213991769547325,
 'national': 0.45251396648044695,
 'largest': 0.3977272727272727,
 'british': 0.4634146341463415,
 'meaning': 0.50625,
 'century': 0.5,
 'musical': 0.375,
 'company': 0.47019867549668876,
 'between': 0.46153846153846156,
 'founded': 0.475177304964539,
 'reports': 0.6170212765957447,
 'character': 0.4,
 'include': 0.4855072463768116,
 'targetblankthisa': 0.6956521739130435,
 'million': 0.44166666666666665,
 'popular': 0.3277310924369748,
 'because': 0.38181818181818183,
 'through': 0.4519230769230769,
 'classic': 0.36893203883495146}

In [87]:
chisq_highest_words = {}
observed_list = sorted(highest_words_observed.items(), key=operator.itemgetter(1), reverse=True)
for v in observed_list:
    chisq_highest_words[v[0]] = (float(v[1]) - jeopardy_high_expected)**2/jeopardy_high_expected
chisq_highest_words

{'targetblankthisa': 0.15506180473691453,
 'targetblankherea': 0.07912409840238467,
 'reports': 0.07543691021949028,
 'meaning': 0.01141606666510203,
 'century': 0.009482401191438967,
 'include': 0.005688474010640689,
 'founded': 0.003572779740223792,
 'company': 0.002728015768679935,
 'british': 0.0017600539105312471,
 'between': 0.0015296474781054607,
 'national': 0.0006471499275248018,
 'through': 0.0006024069327272113,
 'country': 8.648495760203882e-05,
 'million': 8.111048427552069e-05,
 'american': 4.2006424397419386e-05,
 'president': 0.002079234172843375,
 'character': 0.002928579754670537,
 'largest': 0.003313084384818288,
 'capital': 0.005082229878336999,
 'because': 0.006668472052995451,
 'musical': 0.008462132084255116,
 'classic': 0.010237886836776311,
 'popular': 0.026764762028954645}

Excluding the two "targetblank" values, which are some database mistake, one can conclude that "reports" are more frequent in high-valued questions, but it unfortunately doesn't tell much about the topic. "meaning" and "century" are more meaningful, since it refers to synonyms and history respectively.

In [94]:
chances = []
for n in np.arange(0,1000):
    high = 0
    for i in np.arange(0,19999):
        if np.random.rand() <= jeopardy_high_expected:
            high += 1
    chances.append(high/19999)
chance_highest_words = {}
for v in observed_list:
    chance = 0
    for c in chances:
        if float(v[1]) < c:
            chance += 1
    chance_highest_words[v[0]] = chance/10
chance_highest_words

{'targetblankthisa': 0.0,
 'targetblankherea': 0.0,
 'reports': 0.0,
 'meaning': 0.0,
 'century': 0.0,
 'include': 0.0,
 'founded': 0.0,
 'company': 0.0,
 'british': 0.0,
 'between': 0.0,
 'national': 0.0,
 'through': 0.0,
 'country': 4.5,
 'million': 4.9,
 'american': 11.8,
 'president': 100.0,
 'character': 100.0,
 'largest': 100.0,
 'capital': 100.0,
 'because': 100.0,
 'musical': 100.0,
 'classic': 100.0,
 'popular': 100.0}

The list above shows the probability of the frequency of these words happening mostly in high question happening by chance, meaning that, for all words with 0%, there is a preference for them happening in high-valued questions over low-valued ones. Since, even for words that clearly have a higher frequency for high over low-valued questions, the chance of it being asked in a low-valued question is around 50% (like "century"), I wouldn't go with this data to recommend someone to study more this topic, it was good to know as a statistics point of view, but not very useful for the contester 